### ISSUE TITLE
validate does not flag <CR> within lid_reference

### ISSUE THREAD

**Issue by msbentley** [NASA-PDS/validate#401]
## 🐛 Describe the bug
I'm not 100% sure if this is a validate bug, or that something in the IM needs to be better constrained, but I'm assuming for now the former. Products declaring `[lid_reference](https://pds.nasa.gov/datastandards/documents/dd/current/PDS4_PDS_DD_1G00.html#d5e44434)`  (e.g. in `Internal_Reference`) currently allow <CR> according to validate, e.g. this is valid:

```
<Internal_Reference>
    <lid_reference>urn:esa:psa:context:instrument_host:spacecraft.tgo
    </lid_reference>
    <reference_type>is_instrument_host</reference_type>
</Internal_Reference>
```
However, oXygen reports:

```
Value 'urn:esa:psa:context:investigation:mission.em16
                 ' is not facet-valid with respect to pattern 'urn(:[\p{Ll}\p{Nd}\-._]+){3,5}' for type 'lid_reference'.
```

## 📜 To Reproduce
1. Validate a label with the above internal reference

## 🕵️ Expected behavior
I would not expect that carriage return would be allowed within an internal reference, and would expect a validation error.

## 📚 Version of Software Used
validate 2.0.7

## 🖥 System Info
 - OS: ubuntu 
 - Version: 20.04

---

**Comment by qchaupds** [comment]
First inspection of the code reveals something interesting.  The call to node.getTextContent() at line 258 does contains the CR but when the trim() function is applied, the CR is removed.  The bug is identified and can indeed be fixed.

```
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 108 % vi ./src/main/java/gov/nasa/pds/tools/util/LabelUtil.java

241           NodeList nodeList = (NodeList) xPathFactory.newXPath().evaluate(searchPathName,source,XPathConstants.NODESET);
242           LOG.debug("getIdentifiersCommon:context,nodeList.getLength() {},{}",context,nodeList.getLength());
243           LOG.debug("getIdentifiersCommon:context,searchPathName,nodeList.getLength() {},{},{}",context,searchPathName,nodeList.getLength());
244           for (int i = 0; i < nodeList.getLength(); ++i) {
245               NodeList childList = ((Element) nodeList.item(i)).getChildNodes();
246               String singleIdentifier = null;
247               String singleVersion = null;
248               String singleLidvidValue = null;
249               for (int j = 0; j < childList.getLength(); ++j) {
250                   Node node = childList.item(j);
251                   LOG.debug("node.getTextContent().trim() {}",node.getTextContent().trim());
252                   //LOG.debug("node.getTextContent().trim() {}",node.getTextContent().trim());
253                   // Because the tagsList is an array, loop through to check for each tag
254 
255                   for (int kk = 0; kk < tagsList.length; kk++) {
256 
257                       if (node.getNodeName().equals(tagsList[kk]) || node.getNodeName().equals(VERSION_ID_TAG)) {
258                           if (node.getNodeName().equals(tagsList[kk]))
259                               singleIdentifier = node.getTextContent().trim();
260                           if (node.getNodeName().equals(VERSION_ID_TAG))
261                              singleVersion = node.getTextContent().trim();
262                           LOG.debug("getIdentifiersCommon:kk,CONTAINS_CR,TEXTCONTENT {},{},[{}]",kk,node.getTextContent().contains("\n"),node.getTextContent());
263                       }
264                       LOG.debug("getIdentifiersCommon:kk,singleIdentifier,singleVersion {},[{}],[{}]",kk,singleIdentifier,singleVersion);

Running validate on a bundle that does reference checks, we can see that the printout out of line 262 shows CONTAINS_CR as true

{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 129 % grep getIdentifiersCommon:kk,CONTAINS_CR,TEXTCONTENT t1
[main] DEBUG gov.nasa.pds.tools.util.LabelUtil - getIdentifiersCommon:kk,CONTAINS_CR,TEXTCONTENT 1,true,[urn:nasa:pds:kaguya_grs_spectra:document:kgrs_calibrated_spectra ZZZ BBB  
{pds-dev3.jpl.nasa.gov}/home/qchau/sandbox/validate 130 % 

```
